### ISSUE TITLE
Query with relation breaks with soft_delete

### ISSUE THREAD

**Issue by zengxinzhy** [go-pg/pg#1450]
I recently updated my go-pg version to v9 and then met the following problem:
Code example:

```
db := pg.Connect(&pg.Options{
	Database: "postgres",
})
defer db.Close()

type Author struct {
	Id        int
	DeletedAt time.Time `json:"-" pg:",soft_delete"`
}
type Book struct {
	Id       int
	AuthorId int
	Author   *Author
}
for _, model := range []interface{}{(*Author)(nil), (*Book)(nil)} {
	err := db.CreateTable(model, &orm.CreateTableOptions{
		Temp: true,
	})
	if err != nil {
		panic(err)
	}
}
var book = &Book{}
err := db.Model(book).Relation("Author").WherePK().Select()
if err != nil && err != pg.ErrNoRows {
	panic(err)
}
```

I ran the code above, and saw the following error:
```
--- FAIL: TestSoftDelete (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb8 pc=0x13da532]

goroutine 18 [running]:
testing.tRunner.func1(0xc000314400)
	/usr/local/Cellar/go/1.12.5/libexec/src/testing/testing.go:830 +0x392
panic(0x1526ee0, 0x1a22b40)
	/usr/local/Cellar/go/1.12.5/libexec/src/runtime/panic.go:522 +0x1b5
github.com/go-pg/pg/v9/orm.(*Query).appendSoftDelete(0xc00000c1e0, 0xc0003ef000, 0xe4, 0x1000, 0xc0003ef000, 0xe3, 0x1000)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/query.go:1412 +0x82
github.com/go-pg/pg/v9/orm.(*join).appendHasOneJoin(0xc000426000, 0x16656e0, 0xc00000e120, 0xc0003ef000, 0xd7, 0x1000, 0xc00000c1e0, 0xc0003ef08a, 0xc000381898, 0x6, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/join.go:295 +0xaa4
github.com/go-pg/pg/v9/orm.(*selectQuery).AppendQuery.func1(0xc000426000, 0x0, 0x13debf8)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/select.go:92 +0xb2
github.com/go-pg/pg/v9/orm.(*Query)._forEachHasOneJoin(0xc00000c1e0, 0xc0000df798, 0xc000426000, 0x1, 0x1, 0xc000424000, 0xc000406480)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/query.go:962 +0x6c
github.com/go-pg/pg/v9/orm.(*Query).forEachHasOneJoin(0xc00000c1e0, 0xc0000df798, 0xc00000e120, 0xc0003ef000)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/query.go:954 +0x73
github.com/go-pg/pg/v9/orm.(*selectQuery).AppendQuery(0xc00000e100, 0x16656e0, 0xc00000e120, 0xc0003ef000, 0x91, 0x1000, 0x102ea61, 0x15d7b40, 0xc0000df858, 0x0, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/select.go:90 +0x673
github.com/go-pg/pg/v9.appendQuery(0x16656e0, 0xc0002cdc60, 0xc0003ef000, 0x5, 0x1000, 0x15654a0, 0xc00000e100, 0xc0000681b0, 0x1, 0x1, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/messages.go:447 +0x21e
github.com/go-pg/pg/v9.writeQueryMsg(0xc000323a70, 0x16656e0, 0xc0002cdc60, 0x15654a0, 0xc00000e100, 0xc0000681b0, 0x1, 0x1, 0x1a39a60, 0x105f5a0)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/messages.go:427 +0xdd
github.com/go-pg/pg/v9.(*baseDB).simpleQueryData.func1(0xc000323a70, 0x0, 0x0)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:518 +0x75
github.com/go-pg/pg/v9/internal/pool.(*Conn).WithWriter(0xc0004050e0, 0x166f360, 0xc0000c6000, 0x0, 0xc0000dfa50, 0x1365492, 0xbf67d712922395d0)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/internal/pool/conn.go:89 +0xb8
github.com/go-pg/pg/v9.(*baseDB).simpleQueryData(0xc0002fee10, 0x166f360, 0xc0000c6000, 0xc0004050e0, 0x159c040, 0xc000424000, 0x15654a0, 0xc00000e100, 0xc0000681b0, 0x1, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:517 +0x104
github.com/go-pg/pg/v9.(*baseDB).query.func1(0x166f360, 0xc0000c6000, 0xc0004050e0, 0xc0002fee10, 0xc0004050e0)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:284 +0xb4
github.com/go-pg/pg/v9.(*baseDB).withConn(0xc0002fee10, 0x166f360, 0xc0000c6000, 0xc0000dfc58, 0x0, 0x0)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:161 +0xf3
github.com/go-pg/pg/v9.(*baseDB).query(0xc0002fee10, 0x166f360, 0xc0000c6000, 0x159c040, 0xc000424000, 0x15654a0, 0xc00000e100, 0xc0000681b0, 0x1, 0x1, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:283 +0x27c
github.com/go-pg/pg/v9.(*baseDB).QueryContext(...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:264
github.com/go-pg/pg/v9.(*baseDB).queryOne(0xc0002fee10, 0x166f360, 0xc0000c6000, 0x159c040, 0xc000424000, 0x15654a0, 0xc00000e100, 0xc0000681b0, 0x1, 0x1, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:310 +0xb6
github.com/go-pg/pg/v9.(*baseDB).QueryOneContext(0xc0002fee10, 0x166f360, 0xc0000c6000, 0x159c040, 0xc000424000, 0x15654a0, 0xc00000e100, 0xc0000681b0, 0x1, 0x1, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/base.go:306 +0xa7
github.com/go-pg/pg/v9/orm.(*Query).query(0xc00000c1e0, 0x166f360, 0xc0000c6000, 0x16742a0, 0xc000424000, 0x15654a0, 0xc00000e100, 0x1531aa0, 0x0, 0x0, ...)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/query.go:847 +0x155
github.com/go-pg/pg/v9/orm.(*Query).Select(0xc00000c1e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/xinzeng/go/pkg/mod/github.com/go-pg/pg/v9@v9.0.0/orm/query.go:817 +0x122
```

However, if I remove `pg:",soft_delete"` in `Author`, the error disappears. So I think the problem is likely to be related to soft_delete.

**Comment by elliotcourant** [comment]
I believe that particular error is resolved by #1449 but I feel like even without that fix the way the query is being built is kind of odd. WherePk on a default model which would have 0 for the id?

**Comment by vmihailenco** [comment]
On master your example generates following query

``` sql
SELECT "book"."id", "book"."author_id", "author"."id" AS "author__id", "author"."deleted_at" AS "author__deleted_at" FROM "books" AS "book" LEFT JOIN "authors" AS "author" ON ("author"."id" = "book"."author_id") AND "author"."deleted_at" IS NULL WHERE "book"."id" = NULL
```

Seems to be working fine - closing.
